include/quic/endpoint.hpp

```diff
@@ -87,8 +149,8 @@ namespace oxen::quic
         //      a short period of time allowing any lagging packets to be caught
         //
         //      They are indexed by connection ID, storing the removal time as a uint64_t value
-        //
-        std::unordered_map<ConnectionID, std::unique_ptr<Connection>> conns;
+        std::unordered_map<ConnectionID, std::shared_ptr<Connection>> conns;
+
         std::queue<std::pair<ConnectionID, uint64_t>> draining;
```

Comment:
I think this queue should be a `map<expiry, ConnectionID>` instead because I believe the draining period differs per connection (so a low-latency connection could get its drain period unintentionally extended because it gets stuck behind a high-latency drain).